File: src/metrics/metrics.ts
import { BufferedMetricsLogger } from 'datadog-metrics';
import { Config } from '../config/config';

class Metrics {
    private metrics: BufferedMetricsLogger;
    constructor(private config: Config, private prefix: string) {
        if (this.config.metrics.enabled) {
            this.metrics = new BufferedMetricsLogger({
                apiKey: this.config.metrics.api_key,
                appKey: this.config.metrics.app_key,
                prefix: this.prefix + '.',
                host: this.config.metrics.host,
                flushIntervalSeconds: this.config.metrics.flush_interval,
            });
        }
    }

    public gauge(key: string, value: number): void {
        if (this.config.metrics.enabled) {
            this.metrics.gauge(key, value);
        }
    }

    public increment(key: string, value?: number): void {
        if (this.config.metrics.enabled) {
            this.metrics.increment(key, value);
        }
    }

    public histogram(key: string, value: number): void {
        if (this.config.metrics.enabled) {
            this.metrics.histogram(key, value);
        }
    }

    public flush(): void {
        if (this.config.metrics.enabled) {
            this.metrics.flush();
        }
    }
}

File: src/main/java/org/samoxive/safetyjim/metrics/Metrics.java
package org.samoxive.safetyjim.metrics;

import com.timgroup.statsd.NonBlockingStatsDClient;
import com.timgroup.statsd.StatsDClient;

public class Metrics {
    private StatsDClient client;
    private boolean enabled;

    public Metrics(String prefix, String host, int port, boolean enabled) {
        this.enabled = enabled;
        if (enabled) {
            this.client = new NonBlockingStatsDClient(prefix, host, port);
        }
    }

    public void gauge(String key, int value) {
        if (enabled) {
            this.client.gauge(key, value);
        }
    }

    public void increment(String key) {
        if (enabled) {
            this.client.increment(key);
        }
    }

    public void histogram(String key, int value) {
        if (enabled) {
            this.client.histogram(key, value);
        }
    }


}
